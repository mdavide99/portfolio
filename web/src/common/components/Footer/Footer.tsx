import Link from "next/link";
import { FaGithub, FaLinkedinIn } from "react-icons/fa";
import { DarkModeToggle } from "../../elements/DarkMode";

function Footer() {
  return (
    <footer className="container mx-auto px-4">
      <div className="flex justify-between items-center py-4">
        <div>
          <span className="font-light">Copyright Â© Davide Marinotto</span>
        </div>
        <div>
          <span className="font-light">Look inside the hood on </span>
          <Link href={"https://github.com/mdavide99/portfolio"} passHref>
            <a
              rel="noopener noreferrer"
              target="_blank"
              className="underline decoration-2 decoration-blue-500 dark:decoration-fuchsia-500 underline-offset-8 hover:underline-offset-4"
            >
              github
            </a>
          </Link>
        </div>
        <div className="flex">
          <Link href={"https://github.com/mdavide99"} passHref>
            <a
              rel="noopener noreferrer"
              target="_blank"
              className="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
            >
              <FaGithub />
            </a>
          </Link>

          <Link href={"https://www.linkedin.com/in/davide-marinotto/"} passHref>
            <a
              target="_blank"
              className="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
            >
              <FaLinkedinIn />
            </a>
          </Link>

          <DarkModeToggle className="ml-2" />
        </div>
      </div>
    </footer>
  );
}

export { Footer };
