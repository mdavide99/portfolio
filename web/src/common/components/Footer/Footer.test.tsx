import { Footer } from "./Footer";
import { render } from "@testing-library/react";
jest.mock('next/router', () => ({
	__esModule: true,
	useRouter: jest.fn().mockReturnValue({
		route: '',
		pathname: '',
		query: '',
		asPath: '',
	}),
}));

jest.mock('next/link', () => {
	const React = require('react');
	return ({ children, href }: React.PropsWithChildren<{ href: string }>) =>
		React.cloneElement(React.Children.only(children), { href });
});
describe('Footer',()=>{

  it('renders Footer correctly', () => {
		const { container } = render(
		<Footer/>
		);
		expect(container).toMatchSnapshot();
	});
})