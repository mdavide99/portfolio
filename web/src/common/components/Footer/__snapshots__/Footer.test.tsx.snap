// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer correctly 1`] = `
<div>
  <footer
    class="container mx-auto px-4"
  >
    <div
      class="flex justify-between items-center py-4"
    >
      <div>
        <span
          class="font-light"
        >
          Copyright Â© Davide Marinotto
        </span>
      </div>
      <div>
        <span
          class="font-light"
        >
          Look inside the hood on 
        </span>
        <a
          class="underline decoration-2 decoration-blue-500 dark:decoration-fuchsia-500 underline-offset-8 hover:underline-offset-4"
          href="https://github.com/mdavide99/portfolio"
          rel="noopener noreferrer"
          target="_blank"
        >
          github
        </a>
      </div>
      <div
        class="flex"
      >
        <a
          class="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
          href="https://github.com/mdavide99"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
        <a
          class="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
          href="https://www.linkedin.com/in/davide-marinotto/"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
            />
          </svg>
        </a>
        <button
          aria-label="Set Theme"
          class="text-lg ml-2"
        >
          <svg
            class="text-neutral-600 hover:text-blue-600"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
            />
          </svg>
        </button>
      </div>
    </div>
  </footer>
</div>
`;
